### PR TITLE
Staging setup

### PR DESCRIPTION
**Pull request by DanTehrani**
### Summary
- Update GitHub Action workflow to run database migration on two triggers: push to `main` and (re)open of PRs.
   - Migrations will be applied to the production db when there is a push to the `main` branch.
   - Migrations will be applied to the staging db when a PR is (re)opened.
   - (i.e. Upon merge, database migrations will start running for every PR opens and commits to main!)
- Update render.yaml to enable staging env for the merkle tree cron job (more on this below).

Staging db is now populated with test data and is connected to Vercel preview environments: https://nym-l058sqy90-personaelabs.vercel.app/

### Staging env for the frontend
- Previews will point to a single staging db, and prod deployments will point to the prod db. We do so by setting the env var `DATABASE_URL` as:
Screenshot 2023-05-12 at 13 04 44

### Staging env for the Merkle tree cron job
Since we're running the cron job on Render, we use the [Preview Environment of Render](https://render.com/docs/preview-environments) to set up an environment **per PR** for the cron job. Using the singleton staging db as in the frontend is possible, but it's more straightforward to use Preview Environments (which spawns a db per PR) than to point to a single staging db.

### CODE REVIEW

**Comment by cha0sg0d** [comment]
@DanTehrani Looks great! super useful and straightforward for our use cases. 

My one question: what is our flow for inspecting the output for a migration, especially in the case where we have to manually resolve a conflict?

**Comment by cha0sg0d** [comment]
@DanTehrani One other question: Do the migrations run on every push to the pull request branch? Or just when it's opened?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+      - name: Lint
+        run: pnpm lint
+
+      - name: Run tests
+        run: pnpm test
+
+      - name: Run database migrations
+        # Run migrations only on pull requests and pushes to the main branch
+        if: github.event_name == 'pull_request' || github.ref == 'refs/heads/main'
```
Comment (cha0sg0d) [inline .github/workflows/ci.yml:37]: Correct me if I'm wrong, but do we also want to run migrations on pushes to open pull requests?

Comment (DanTehrani) [reply .github/workflows/ci.yml:37]: @cha0sg0d 
Actually yes, added the [synchronize trigger](https://github.com/orgs/community/discussions/24567) as well: d5ba46ae7292a8f5af8e0e8c01ca014296b01555

**Comment by lsankar4033** [comment]
lgtm!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (cha0sg0d) [reply .github/workflows/ci.yml:37]: Great, good to merge now.

**Pull request merged**